native/swift/Sources/wordpress-api/Pagination.swift

```diff
@@ -0,0 +1,135 @@
+import WordPressAPIInternal
+
+public protocol PaginatableResponse {
+    associatedtype ParamsType
+    associatedtype DataType
+
+    var nextPageParams: ParamsType? { get }
+    var prevPageParams: ParamsType? { get }
+
+    var data: [DataType] { get }
+}
+
+public protocol PaginationAwareExecutor {
+    associatedtype EditContextResponseType: PaginatableResponse
+    associatedtype ViewContextResponseType: PaginatableResponse
+    associatedtype EmbedContextResponseType: PaginatableResponse
+
+    /// Known function signatures for Request Executors
+    func listWithEditContext(params: EditContextResponseType.ParamsType) async throws -> EditContextResponseType
+    func listWithViewContext(params: ViewContextResponseType.ParamsType) async throws -> ViewContextResponseType
+    func listWithEmbedContext(params: EmbedContextResponseType.ParamsType) async throws -> EmbedContextResponseType
+
+    /// Generated implementation
+    func paginatedWithEditContext(
+        params: EditContextResponseType.ParamsType
+    ) async throws -> [EditContextResponseType.DataType]
+
+    func paginatedWithViewContext(
+        params: ViewContextResponseType.ParamsType
+    ) async throws -> [ViewContextResponseType.DataType]
+
+    func paginatedWithEmbedContext(
+        params: EmbedContextResponseType.ParamsType
+    ) async throws -> [EmbedContextResponseType.DataType]
+}
+
+extension PaginationAwareExecutor {
+    /// Fetches all objects from all pages
+    ///
+    /// This method waits until all objects have been downloaded then returns the results. This can have
+    /// unexpected memory and time implications.
+    public func paginatedWithEditContext(
+        params: EditContextResponseType.ParamsType
+    ) async throws -> [EditContextResponseType.DataType] {
+        var workingResponse = try await self.listWithEditContext(params: params)
+        var allObjects: [EditContextResponseType.DataType] = workingResponse.data
+
+        while let nextPageParams = workingResponse.nextPageParams {
+            workingResponse = try await self.listWithEditContext(params: nextPageParams)
+            allObjects.append(contentsOf: workingResponse.data)
+        }
```

Comment:
Nitpick: using `do-while` to avoid duplicating `listWithEditContext` and `appending(contentsOf:...)` calls.

Reply:
Ah, that is nicer. Done in 098c1e8.